resource/resource.go

```diff
@@ -342,6 +342,25 @@ func (r *Resource) find(ctx context.Context, q *query.Query, forceTotal bool) (l
 	return
 }
 
+// Count implements Storer interface.
+func (r *Resource) Count(ctx context.Context, q *query.Query) (n int, err error) {
+	if LoggerLevel <= LogLevelDebug && Logger != nil {
+		defer func(t time.Time) {
+			found := n
+			Logger(ctx, LogLevelDebug, fmt.Sprintf("%s.Count(...)", r.path), map[string]interface{}{
+				"duration": time.Since(t),
+				"found":    found,
+				"error":    err,
+			})
+		}(time.Now())
+	}
+	if err = r.hooks.onFind(ctx, q); err == nil {
```

Comment:
Should be an onCount handler or no handler.

Reply:
As I understand it, HEAD must behave like GET, just not to get the result, so I reused the handlers. Having no handlers on HEAD will prevent implementing security policies, like preventing a request of getting total resource count, etc.

Reply:
I would tend to agree with @Dragomir-Ivanov. There are other cases than _authentication_ hooks for the Find logic, such as manipulating the fields, but I suppose those would just work anyway if the response is empty.

Reply:
I am comfortable with adding separate hooks for HEAD, as well. I think that gives the maximum flexibility. Users then can use shared code in `onFind/OnFound` and `onHead`. I guess no past-tense hook is needed, since there is no items to return.

Reply:
@rs Any comment on the hooks issue?